File: webstore/webstore-java/src/main/java/com/renan/webstore/business/admin/CategoryService.java
package com.renan.webstore.business.admin;

import com.renan.webstore.business.BusinessException;
import com.renan.webstore.controller.form.admin.CategoryForm;
import com.renan.webstore.model.Category;

import java.util.List;

public interface CategoryService {
    
    List<Category> getCategories() throws BusinessException;
    
    String createCategory(CategoryForm form) throws BusinessException;
    
    void updateCategory(CategoryForm form, String id) throws BusinessException;
    
    void deleteCategory(String id) throws BusinessException;

}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/business/CategoryService.kt
package com.renan.webstore.business

import com.renan.webstore.controller.admin.form.CategoryForm
import com.renan.webstore.dto.CategoryDTO

interface CategoryService {

    @Throws(BusinessException::class)
    fun getCategories(): List<CategoryDTO>

    @Throws(BusinessException::class)
    fun createCategory(form: CategoryForm): Int

    @Throws(BusinessException::class)
    fun updateCategory(form: CategoryForm, id: Int)

    @Throws(BusinessException::class)
    fun deleteCategory(id: Int)

}